File: part-0/java/Main.java
public class Main {
    // Worry about this!!
    private static void runTests() {
        // If you want to see how this works, look at PrimitiveTestFramework.java
        PrimitiveTestFramework t = new PrimitiveTestFramework();

        // Examples:
        t.assertTrue("true", true);
        t.assertEqual("1 + 1 == 2", 1 + 1, 2);

        // Try writing your own tests!
        locationTests(t);
        boardTests(t);

        t.endTests();
    }

    private static void locationTests(PrimitiveTestFramework t) {
        t.assertEqual("Two locations should be equal if they have the same index", new Location(65), new Location(65));

        t.assertEqual("Location a1 should be index 91", new Location('a', 1), new Location(91));
        t.assertEqual("Location a8 should be index 21", new Location('a', 8), new Location(21));
        t.assertEqual("Location h8 should be index 28", new Location('h', 8), new Location(28));
        t.assertEqual("Location e5 should be index 55", new Location('e', 5), new Location(55));

        t.assertTrue("Location with index 4 should not be valid", !new Location(4).isValid());
        t.assertTrue("Location with index 10 should not be valid", !new Location(10).isValid());
        t.assertTrue("Location with index 19 should not be valid", !new Location(19).isValid());
        t.assertTrue("Location with index 50 should not be valid", !new Location(50).isValid());
        t.assertTrue("Location with index 101 should not be valid", !new Location(101).isValid());

        t.assertTrue("Location with index 21 should be valid", new Location(21).isValid());
        t.assertTrue("Location with index 28 should be valid", new Location(28).isValid());
        t.assertTrue("Location with index 55 should be valid", new Location(55).isValid());

        // TODO: Add more tests here!
    }

    private static void boardTests(PrimitiveTestFramework t) {
        Board board = Board.startingPosition();
        t.assertEqual("Starting position a1", board.getPiece(new Location('a', 1)), Piece.WhiteRook);
        t.assertEqual("Starting position e8", board.getPiece(new Location('e', 8)), Piece.BlackKing);
        t.assertEqual("Starting position d3", board.getPiece(new Location('a', 1)), Piece.Empty);
        t.assertEqual("Starting position b7", board.getPiece(new Location('a', 1)), Piece.BlackPawn);

        // TODO: Add more tests here!
    }

    // Don't worry about this!!
    public static void main(String[] args) {
        if (args.length != 0 && args[0].equals("test")) {
            runTests();
        } else {
            runProgram();
        }
    }

    private static void runProgram() {
        System.out.println("We'll get to this soon!");
        System.out.println("If you want to run tests, run with the argument \"test\".");
    }
}

File: part-0/python/main.py
from primitive_test_framework import PrimitiveTestFramework
from location import Location
from board import Board
from piece import Piece


# Worry about this!!
def run_tests():
    t = PrimitiveTestFramework()
    t.assert_true("True is true", False)
    t.assert_equal("1 + 1 = 2", 1 + 1, 2)

    # Try writing your own tests!
    location_tests(t)
    board_tests(t)

    t.end_tests()


def location_tests(t: PrimitiveTestFramework):
    t.assert_equal("Two locations should be equal if they have the same index", Location(
        65), Location(65))

    t.assert_equal("Location a1 should be index 91",
                   Location.from_file_rank('a', 1), Location(91))
    t.assert_equal("Location a8 should be index 21",
                   Location.from_file_rank('a', 8), Location(21))
    t.assert_equal("Location h8 should be index 28",
                   Location.from_file_rank('h', 8), Location(28))
    t.assert_equal("Location e5 should be index 55",
                   Location.from_file_rank('e', 5), Location(55))

    t.assert_equal("Location a1 should be in file a",
                   Location.from_file_rank('a', 1).file(), "a")
    t.assert_equal("Location g3 should be in file g",
                   Location.from_file_rank('g', 3).file(), "g")
    t.assert_equal("Location a1 should be in rank 1",
                   Location.from_file_rank('a', 1).rank(), 1)
    t.assert_equal("Location h8 should be in rank 8",
                   Location.from_file_rank('h', 8).rank(), 8)

    t.assert_true("Location with index 4 should not be valid",
                  not Location(4).is_valid())
    t.assert_true("Location with index 10 should not be valid",
                  not Location(10).is_valid())
    t.assert_true("Location with index 19 should not be valid",
                  not Location(19).is_valid())
    t.assert_true("Location with index 50 should not be valid",
                  not Location(50).is_valid())
    t.assert_true("Location with index 101 should not be valid",
                  not Location(101).is_valid())

    t.assert_true("Location with index 21 should be valid",
                  Location(21).is_valid())
    t.assert_true("Location with index 28 should be valid",
                  Location(28).is_valid())
    t.assert_true("Location with index 55 should be valid",
                  Location(55).is_valid())

    # Try writing your own tests!


def board_tests(t: PrimitiveTestFramework):
    t.assert_equal("Starting position a1", Board.starting_position().get_piece(
        Location.from_file_rank('a', 1)), Piece(Piece.WHITE_ROOK))
    t.assert_equal("Starting position e8", Board.starting_position().get_piece(
        Location.from_file_rank('e', 8)), Piece(Piece.BLACK_KING))
    t.assert_equal("Starting position d3", Board.starting_position().get_piece(
        Location.from_file_rank('d', 3)), Piece(Piece.EMPTY))
    t.assert_equal("Starting position b7", Board.starting_position().get_piece(
        Location.from_file_rank('b', 7)), Piece(Piece.BLACK_PAWN))
    # Try writing your own tests!

# Don't worry about this!


def run_program():
    print("We'll do this later!")
    print("If you want to run the tests, run this file with the test argument.")


if __name__ == "__main__":
    import sys
    if len(sys.argv) > 1 and sys.argv[1] == "test":
        run_tests()
    else:
        run_program()
